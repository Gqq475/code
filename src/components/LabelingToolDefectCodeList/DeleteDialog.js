import React from 'react'
import classes from './Dialog.scss'
import { Button, Dialog } from '@blueprintjs/core'

type Props = {
  id: Number,
  isOpen: boolean,
  toggleDialog: Function
};

export class DeleteDialog extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit () {
    this.props.toggleDialog()
  }

  render () {
    return (
      <div className={classes['Dialog-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>提示</h6>
          </div>
          <div className={classes.dialogBody}>
            <br />
            <p className={classes.cn}>{`是否删除id为${this.props.id}的这条数据?`}</p>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                {_.CANCEL}
              </Button>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                {_.OK}
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

export default DeleteDialog
