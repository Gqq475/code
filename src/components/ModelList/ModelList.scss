.ModelList-container {
  // width: 80%;
  border-radius: 5px;
  padding: 0 0 10px 0;
  position: relative;
  // height: 360px;
  border: 1px solid #0087DC;

  .ag-table{
    height: 330px;
  }

  .cell {
    border-radius: 5px;
    color: #fff;
    background: #9B9B9B;
    display: inline-block;
    width: 100px;
    overflow: hidden;
    font-size: 12px
  }

  .progress {
    width: initial;
    margin-left: 5px;
    padding: 0 5px;
    background: #2887C3;
    border-radius: 0;
  }
  .cn{
    text-align: center;
  }
  .status{
    padding: 5px 10px;
    border-radius: 5px 0 5px 0;
    color: #fff;
    background: #9B9B9B;
    &.status-s{
      background: #7ED321;
    }

    &.status-d{
      background: #D0021B;
    }

    &.status-w{
      background: #F5A623;
    }

    &.status-p{
      background: #4A90E2;
    }
  }
}
