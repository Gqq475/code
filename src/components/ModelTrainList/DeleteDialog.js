import React from 'react'
import classes from './Dialog.scss'
import { Button, Dialog } from '@blueprintjs/core'
import OurToaster from 'components/OurToaster'
import { connect } from 'react-redux'
const qca = __QCA__

type Props = {
  id: Number,
  api: Object,
  isOpen: boolean,
  toggleDialog: Function,
  token: String
};

export class DeleteDialog extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit () {
    // const url = `${qca}/ojs/service`
    const url = `${qca}/offline/version_control/model_version`
    const delData = {}
    delData.model_id = this.props.id.toString()
    fetch(url, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': this.props.token
      },
      body: JSON.stringify(delData)
    })
    .then(response => response.json())
    .then(data => {
      if (data.code === 200000) {
        this.props.api.removeItems(this.props.api.getSelectedNodes())
      } else {
        OurToaster.show({message: '删除失败！'})
      }
    })

    this.props.toggleDialog()
  }

  render () {
    return (
      <div className={classes['Dialog-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>提示</h6>
          </div>
          <div className={classes.dialogBody}>
            <br />
            <p className={classes.cn}>{`是否删除id为${this.props.id}的这条数据?`}</p>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                {_.CANCEL}
              </Button>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                {_.OK}
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(DeleteDialog)
