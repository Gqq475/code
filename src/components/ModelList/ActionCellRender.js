import React from 'react'
import DeleteDialog from './DeleteDialog'
import classes from './ModelList.scss'

type Props = {
  data: Object,
  api: Object
};

export class ActionCellRender extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.state = {
      isDeleteBtnOpen: false
    }
  }

  onDelete () {
    this.setState({
      isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    })
  }

  render () {
    return (
      <div className={classes.cn}>
        <button onClick={this.onDelete}
          className={'pt-button pt-minimal pt-icon-delete'}
          title='Delete' />
        <DeleteDialog
          api={this.props.api}
          id={this.props.data.id}
          isOpen={this.state.isDeleteBtnOpen}
          toggleDialog={this.onDelete} />
      </div>
    )
  }
}

export default ActionCellRender
