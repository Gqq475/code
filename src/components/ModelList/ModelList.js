import React from 'react'
import classes from './ModelList.scss'
import {AgGridReact} from 'ag-grid-react'
import ActionCellRender from './ActionCellRender'
import Utils from '../../Utils.js'
import { connect } from 'react-redux'

type Props = {
  setDetail: Function,
};

export class ModelList extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    const columnDefs = [
      {headerName: '模型编号', width: 120, field: 'id', suppressMovable: true, cellClass: classes.cn},
      {headerName: '模型名称', width: 120, field: 'name', suppressMovable: true, cellClass: classes.cn},
      {headerName: '状态', width: 120, field: 'status', suppressMovable: true, cellClass: classes.cn},
      {headerName: '产品', width: 120, field: 'productCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '站点', width: 120, field: 'operationCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '线别', width: 120, field: 'lineCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '机台', width: 120, field: 'eqCode', suppressMovable: true, cellClass: classes.cn},
      {headerName: '创建时间', width: 180, suppressMovable: true, field: 'createTime', cellClass: classes.cn},
      {headerName: '操作',
        suppressMovable: true,
        width: 80,
        field: 'status',
        suppressSorting: true,
        cellRendererFramework: ActionCellRender
      }
    ]
    this.onRowClicked = this.onRowClicked.bind(this)
    this.datasource = this.datasource.bind(this)
    this.data = {}
    this.state = {
      columnDefs
    }
  }

  onRowClicked (e) {
    this.bypassRender = true
    this.props.setDetail(e.data)
  }

  shouldComponentUpdate () {
    if (this.bypassRender) {
      this.bypassRender = false
      return false
    }

    return true
  }

  datasource () {
    let smIns = this
    return {
      getRows: function (params) {
        let pageSize = 10
        let pageNo = params.endRow / pageSize
        let url = Utils.format(smIns.props.data.mvTemplate, [pageNo, pageSize])
        fetch(url, {
          headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'Authorization': smIns.props.token
          }
        })
        .then(response => response.json())
        .then((data) => {
          params.successCallback(data.result.models, data.result.totalCount)
        })
      }
    }
  }

  render () {
    return (
      <div className={classes['ModelList-container']}>
        <section className={`${classes['ag-table']} ag-fresh`}>
          <AgGridReact
            enableSorting
            sortingOrder={['desc', 'asc']}
            rowSelection='single'
            rowModelType='pagination'
            paginationPageSize={10}
            columnDefs={this.state.columnDefs}
            onRowClicked={this.onRowClicked}
            datasource={this.datasource()}
          />
        </section>
      </div>
    )
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(ModelList)
