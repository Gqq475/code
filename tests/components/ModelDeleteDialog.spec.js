import React from 'react'
import { DeleteDialog } from 'components/ModelList/DeleteDialog'

describe('(Component) DeleteDialog', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <DeleteDialog />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
