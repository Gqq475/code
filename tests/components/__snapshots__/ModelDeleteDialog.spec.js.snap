// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DeleteDialog render normally 1`] = `
<div
  className="Dialog-container"
>
  <Dialog
    className="dialog"
    isOpen={false}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        提示
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        是否删除id为undefined的这条数据?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
        />
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        />
      </div>
    </div>
  </Dialog>
</div>
`;
